geometry/src/vector.rs

```diff
@@ -116,6 +116,15 @@ impl Vector {
 
         self / magnitude
     }
+
+    /// Returns the projection of this vector onto another vector
+    pub fn project_onto(self, other: Vector) -> Vector {
+        let zero_vector = Default::default();
+        if self == zero_vector || other == zero_vector {
```

Comment:
We could get rid of the explicit `return` here if we make this an if / else.

Reply:
Addressed in https://github.com/myelin-ai/engine/pull/58/commits/f8959c3aaeabeee85dccc1fabf833e15b177a412